hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operand/OperandCheckerProgram.java

```diff
@@ -32,9 +40,11 @@ public OperandCheckerProgram(OperandChecker... checkers) {
     public boolean check(HazelcastCallBinding callBinding, boolean throwOnFailure) {
         boolean res = true;
 
+        boolean hasAssignment = hasAssignment(callBinding.getCall());
         for (int i = 0; i < checkers.length; i++) {
-            boolean checkerRes = checkers[i].check(callBinding, false, i);
+            OperandChecker checker = hasAssignment ? findCheckerByParamName(callBinding, i) : checkers[i];
```

Comment:
Can't we always find the checker by param name? Isn't it the more general way?

Reply:
When one uses positional arguments then there are no names (it's not an argument assignment) we could match params by.

Reply:
But the what about mixed named and non-named operands? The `hasAssignment` looks at all the operands, if any is named, we extract all of them by name.

Would also be nice if we could add a test in imdg - both for all named and mixed arguments.

Reply:
Calcite (we) does not support mixed operands.

Reply:
Calcite (we) does not support mixed operands.